cli/ops/worker_host.rs

```diff
@@ -147,28 +194,41 @@ fn op_create_worker(
   let specifier = args.specifier.clone();
   let has_source_code = args.has_source_code;
   let source_code = args.source_code.clone();
-  let args_name = args.name;
+  let maybe_worker_name = args.name;
+  let use_deno_namespace = args.use_deno_namespace;
   let parent_state = state.clone();
   let state = state.borrow();
   let global_state = state.global_state.clone();
-  let permissions = state.permissions.clone();
+
+  // TODO(bartlomieju): assert that doesn't escalate parent permissions
+  let permissions = if let Some(p) = args.permissions {
+    DenoPermissions {
+      allow_read: PermissionState::from(p.read),
+      allow_write: PermissionState::from(p.write),
+      allow_net: PermissionState::from(p.net),
+      allow_env: PermissionState::from(p.env),
+      allow_run: PermissionState::from(p.run),
+      allow_plugin: PermissionState::from(p.plugin),
+      allow_hrtime: PermissionState::from(p.hrtime),
+      ..Default::default()
+    }
+  } else {
+    state.permissions.clone()
+  };
   let referrer = state.main_module.to_string();
   drop(state);
 
   let module_specifier =
     ModuleSpecifier::resolve_import(&specifier, &referrer)?;
```

Comment:
Workers are very much like dynamic imports - they are created during runtime, so proper permissions for module loading should be checked when new worker is created.